azure-pipelines/azure-pipelines.yml

```diff
@@ -645,3 +645,140 @@ stages:
           --build-arg PROJECT_ROOT="$(System.DefaultWorkingDirectory)" $(System.DefaultWorkingDirectory) \
           -t edgehub/mockdevice-opcua:$(release_tag) --push
         displayName: build edgehub/mockdevice-opcua
+
+- stage: docker_build_shifu_demo_push
+  condition: and(succeeded(), variables['build.retainedByRelease'])
+  jobs:
+  - job: docker_push_shifu_demo
+    steps:
+    - task: DockerInstaller@0
+      displayName: Docker Installer
+      inputs:
+        dockerVersion: 20.10.9
+        releaseType: stable
+
+    - script: |
+        set -e
+        bash test/scripts/deviceshifu-demo-aio.sh build_demo amd64
+        bash test/scripts/deviceshifu-demo-aio.sh build_demo arm64
+        ls -al
+      displayName: 'Run the build script'
+    - script: |
+        set -e
+        mkdir testdir && tar -xvf shifu_demo_aio_linux_amd64.tar -C testdir && cd testdir
+        chmod +x scripts/deviceshifu-demo-aio.sh && sudo ./scripts/deviceshifu-demo-aio.sh run_demo
+      displayName: 'test run_demo'
+    - script: |
+        set -e
+        sudo kubectl run nginx --image=nginx:1.21 -n deviceshifu
+        sudo kubectl wait --for=condition=Available deploy/shifu-crd-controller-manager -n shifu-crd-system --timeout=150s
+        sudo kubectl wait --for=condition=Available deploy/deviceshifu-agv-deployment -n deviceshifu --timeout=150s
+        sudo kubectl wait --for=condition=Available deploy/agv -n devices --timeout=150s
+        sudo kubectl wait --for=condition=Ready pod/nginx -n deviceshifu --timeout=150s
+        sudo kubectl exec -it -n deviceshifu nginx -- curl http://deviceshifu-agv/get_position
+      displayName: 'test run_demo result'
+    - task: CopyFilesOverSSH@0
+      condition: and(succeeded(), eq(variables['build.sourceBranch'], 'refs/heads/main'))
```

Comment:
need to change the sourceBranch condition, demo should run for each tagged version.
same for all occurrences in pipeline